src/BSON/Document.c

```diff
@@ -591,13 +598,25 @@ zval* php_phongo_document_read_dimension(phongo_compat_object_handler_type* obje
 	intern = Z_OBJ_DOCUMENT(PHONGO_COMPAT_GET_OBJ(object));
 
 	if (Z_TYPE_P(offset) != IS_STRING) {
-		phongo_throw_exception(PHONGO_ERROR_RUNTIME, "Could not find key of type \"%s\" in BSON document", PHONGO_ZVAL_CLASS_OR_TYPE_NAME_P(offset));
-		return &EG(uninitialized_zval);
+		if (type != BP_VAR_IS) {
+			phongo_throw_exception(PHONGO_ERROR_RUNTIME, "Could not find key of type \"%s\" in BSON document", PHONGO_ZVAL_CLASS_OR_TYPE_NAME_P(offset));
+			return &EG(uninitialized_zval);
+		}
+
+		ZVAL_NULL(rv);
+
+		return rv;
```

Comment:
Should improve readability.

```suggestion
		if (type == BP_VAR_IS) {
			ZVAL_NULL(rv);
			return rv;
		}

		phongo_throw_exception(PHONGO_ERROR_RUNTIME, "Could not find key of type \"%s\" in BSON document", PHONGO_ZVAL_CLASS_OR_TYPE_NAME_P(offset));

		return &EG(uninitialized_zval);
```

Reply:
Updated 👍